quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HttpPropertyMappers.java

```diff
@@ -23,144 +19,99 @@ private HttpPropertyMappers(){}
 
     public static PropertyMapper[] getHttpPropertyMappers() {
         return new PropertyMapper[] {
-                builder().from("http-enabled")
+                fromOption(HttpOptions.httpEnabled)
                         .to("quarkus.http.insecure-requests")
-                        .defaultValue(Boolean.FALSE.toString())
-                        .transformer(HttpPropertyMappers::getHttpEnabledTransformer)
```

Comment:
Why change the transformer? Is it related to the scope of this PR? Is a method reference changing anything?

If not, I would avoid any changes at this point in favor of making the review process easier.

Reply:
Unfortunately, with the new types around, the compiler cannot infer the correct types, and we need an explicit `BiFunction`, open to change to a different encoding if you prefer.